pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -252,6 +295,31 @@ func (r *ReconcileOneAgent) reconcileRollout(logger logr.Logger, instance *dynat
 	return updateCR, nil
 }
 
+func (r *ReconcileOneAgent) determineOneAgentPhase(instance *dynatracev1alpha1.OneAgent) (bool, error) {
+	// Check if this DaemonSet already exists
+	var phaseChanged bool
+	dsActual := &appsv1.DaemonSet{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, dsActual)
+	if err != nil && !errors.IsNotFound(err) {
+		phaseChanged = instance.Status.Phase != dynatracev1alpha1.Error
+		instance.Status.Phase = dynatracev1alpha1.Error
+		return phaseChanged, err
+	}
+
+	// Case: Daemonset exists
+	if !errors.IsNotFound(err) {
```

Comment:
this is already being checked for, at 303.. 

```
if err != nil && !errors.IsNotFound(err)
```
and
```
if !errors.IsNotFound(err) {}
```
are basically same conditions.

Reply:
Hmm, they are different: the first one says "any error that is not NotFound" and the second "the error is not NotFound" (i.e., exists), the leftover case is that the error is NotFound (i.e., doesn't exist)

Reply:
// Check if this DaemonSet already exists
	// Case: Daemonset exists
￼How are they different :confused: 
I would say split it logically if they are really that different, but imo, it does not seem so.